patches/tModLoader/Terraria/ModLoader/ModAccessorySlot.cs

```diff
@@ -0,0 +1,86 @@
+using Microsoft.Xna.Framework;
+using Terraria.ModLoader.Default;
+using Terraria.UI;
+
+//NOTE: Does not fully support: ItemLoader.GetGamepadInstructions and related gamepad stuff.
+//TODO: Documentation?
+
+namespace Terraria.ModLoader
+{
+	/// <summary>
+	/// A ModAccessorySlot instance represents a net new accessory slot instance. You can store fields in the ModAccessorySlot class. 
+	/// </summary>
+	public abstract class ModAccessorySlot : ModType
+	{
+		internal int type;
+		internal int index;
+		public AccessorySlotLoader Loader => LoaderManager.Get<AccessorySlotLoader>();
+
+		// Properties to preset a location for the accessory slot
```

Comment:
There should probably be get+set properties for `Item`, `VanityItem`, `Dye` and `ShowVisuals`